src/priorityQueue.js

```diff
@@ -0,0 +1,67 @@
+'use strict'
+
+export default class PriorityQueue {
+  constructor() {
+    this.storage = [],
+    this.end = 0
+  }
+
+  length() {
+    return this.end
+  }
+
+  enqueue( item, priority ) {
+    const newEntry = { [priority]: item  }
```

Comment:
This is an interesting way to store the priority.  It's space efficient.  Another way is `{ priority: priority, item: item}`